### DISCUSSION TITLE
GeoJSON type for editable layers to match deck.gl v9

### DISCUSSION THREAD

**Discussion by dm2x46** [visgl/deck.gl-community#106] [General]
Hi @ibgreen,

I hope this message finds you well. I wanted to open a discussion about the GeoJSON type used in editable layers. Currently, the GeoJSON type used in editable layers seems to differ from the one used in deck.gl v9. Are there any plans to update the GeoJSON type for editable layers to match with the GeoJSON type implementation in deck.gl v9?

I appreciate your time and look forward to hearing your thoughts on this matter.

Thank you!

**Comment by ibgreen** [comment]
@dm2x46 Thanks for your input. I took a quick look and I think it should be possible to align the GeoJSON types with deck.gl and loaders.gl, perhaps by importing types from the `geojson` module.

Unfortunately there are no dedicated maintainers for the frameworks in deck.gl-community who would pick this up, so feature changes are done mostly by community PRs.

If you are able to put up a PR I would be willing to review and help you get it ready to land.

**Reply by dm2x46** [reply]
Thank you for your response, @ibgreen.

Following up on your suggestion, I've been working on aligning the GeoJSON types in deck.gl-community. My approach has been to remove the geojson-types and instead add @types/geojson package. However, I've encountered an issue during this process.

When I added `@types/geojson` to the `modules/editable-layers` folder with this command `yarn add @types/geojson`, I received the error:

![image](https://github.com/user-attachments/assets/6bcfb4c2-ff54-48cf-b357-53da94b26ec3)

I'm figuring out how to resolve this error while maintaining the goal of aligning the GeoJSON types with deck.gl and loaders.gl. Would you have any insights on how to address this issue?

I'll try to see this through and create a PR once we've found a viable solution.

Thank you again for your support.

**Reply by felixpalmer** [reply]
It looks like you're on Windows (which I'm not), so I can't help with this specific error. Are you able to run `yarn build` successfully without making any changes to the repo? You might want to consult https://github.com/visgl/deck.gl/blob/master/CONTRIBUTING.md to check your dev environment is correctly setup

**Reply by ibgreen** [reply]
I just tried to drop "gl" dependency which should work if we are on the latest ocular-dev-tools but noticed that graph-layers includes gpu.js which includes "gl"

You could try to set resolutions to force "gl" to latest version and see if that helps.

And you need to be on windows subsystem for Linux as @felixpalmer mentioned.

**Reply by giangbv1984** [reply]
Hi @ibgreen. The deck.gl v9 use the geojson ("@types/geojson": "^7946.0.8"). I can update it. Can you guide the git rules?Thanks

**Comment by ibgreen** [comment]
This should be published in `@deck.gl-community/editable-layers@9.0.2`. 
- Please reopen if that is not the case. 
- Please start new discussions if there are other follow-up issues.
